onnxscript/function_libs/torch_lib/ops/core.py

```diff
@@ -6257,6 +6257,24 @@ def aten_tanh(self: TFloat) -> TFloat:
     return op.Tanh(self)
 
 
+@torch_op("aten::tensor.bool")
+def aten_tensor_bool(self: bool, dtype: int) -> TensorType:
```

Comment:
Should we modify the TensorType to one of TypeVar we have?

Reply:
it's `dtype` dependent and is equivalent to the scope of `onnx::Cast`, what should we use? I mimic-ed what `prims_convert_element_type`, aka `cast` did.


```def prims_convert_element_type(a: TensorType, dtype: int) -> TensorType:```

Reply:
We can probably use the underlying type for `TTensor`. (not the typevar since the input and output are not constraint to the same type) Right now it is `_TensorType`. We can give it a better name to avoid collision.

Reply:
Since `TensorType` may accept String as well which is not we want, but it is not a big deal